syntax = "proto3";

option go_package = "pb/game";

message ChangeSceneInfoPBComponent
{

  enum eChangeGsType
  {
      eSameGs = 0;//同服务器间切换
      eDifferentGs = 1;//不同服务器间切换
  };
  
  enum eChangeSceneState  
  {
	ePendingLeave = 0;
	eLeaving = 1;
	eWaitingEnter =2;
	eEnterSucceed = 3;
	eGateEnterSucceed = 4;
	eDone = 5;
  };

  
  uint32 scene_confid = 1;
  uint32 guid = 2;//场景id
  uint32 dungen_confid = 3;
  uint32 mirror_confid = 4;
  uint32 change_sequence = 5;
  uint32 change_gs_type = 6;
  uint64 change_time = 7;//设置超时，超时代表服务器崩溃了,或者网络延时
  bool ignore_full = 8;//忽略人数满了
  bool processing = 9;
  uint32 state = 10;
  uint32 from_zone_id = 11;
  uint32 to_zone_id = 12;   // ✨ 一定要放入结构里！
  bool is_cross_zone = 13; // 是否跨 Zone
}